### PR TITLE
PATCH on Chassis/Power

### PR DESCRIPTION
**Pull request by ehaligow**
Added handling PATCH on Power in svc-systems, svc-device-manager and svc-api.

### CODE REVIEW

File: svc-api/rpc/chassis.go [hunk 1]
```diff
@@ -113,3 +112,20 @@ func UpdateChassis(req chassisproto.UpdateChassisRequest) (*chassisproto.GetChas
 	}
 	return resp, nil
 }
+
+//UpdateChassisResource will do the rpc call to update a chassis resource
+func UpdateChassisResource(req chassisproto.UpdateChassisResourceRequest) (*chassisproto.GetChassisResponse, error) {
+	conn, err := services.ODIMService.Client(services.Systems)
+	if err != nil {
+		return nil, fmt.Errorf("Failed to create client connection: %v", err)
```
Comment (MaciejMis) [inline svc-api/rpc/chassis.go:120]: Use the same error description as error below - "error: failed to create client connection" or remove "error:" from error below

File: svc-device-manager/rest/redfish/client.go [hunk 2]
```diff
@@ -143,6 +143,39 @@ func (h *HttpClient) Post(uri string, requestBody []byte) (*http.Response, error
 	return resp, nil
 }
 
+func (h *HttpClient) Patch(uri string, requestBody []byte) (*http.Response, error) {
```
Comment (MaciejMis) [inline svc-device-manager/rest/redfish/client.go:146]: Please add a comment what this func do (like in previous functions in this file)

File: svc-systems/rpc/chassis.go [hunk 3]
```diff
@@ -59,6 +59,24 @@ type ChassisRPC struct {
 	CreateHandler        *chassis.Create
 }
 
+func (cha *ChassisRPC) UpdateChassisResource(_ context.Context, req *chassisproto.UpdateChassisResourceRequest) (*chassisproto.GetChassisResponse, error) {
```
Comment (MaciejMis) [inline svc-systems/rpc/chassis.go:62]: Remove context from func signature (looks like its not used)

**Review by MaciejMis — CHANGES_REQUESTED** [review]

File: svc-api/handle/chassis.go [hunk 4]
```diff
@@ -229,3 +230,35 @@ func (chassis *ChassisRPCs) DeleteChassis(ctx iris.Context) {
...
+
+
+// UpdateChassisResource updates an existing resource under chassis
+func (chassis *ChassisRPCs) UpdateChassisResource(ctx iris.Context) {
+	defer ctx.Next()
+	requestBody := new(json.RawMessage)
+	e := ctx.ReadJSON(requestBody)
+	if e != nil {
+		errorMessage := "error while trying to read obligatory json body: " + e.Error()
+		log.Println(errorMessage)
```
[the remark this answers is not in the document]
Comment (ehaligow) [reply svc-api/handle/chassis.go]: Done.

File: svc-api/handle/chassis.go [hunk 5]
```diff
@@ -229,3 +230,35 @@ func (chassis *ChassisRPCs) DeleteChassis(ctx iris.Context) {
...
+		response := common.GeneralError(http.StatusBadRequest, response.MalformedJSON, errorMessage, nil, nil)
+		common.SetResponseHeader(ctx, response.Header)
+		ctx.StatusCode(http.StatusBadRequest)
+		ctx.JSON(&response.Body)
+		return
+	}
+	rr, rerr := chassis.UpdateChassisResourceRPC(chassisproto.UpdateChassisResourceRequest{
```
[the remark this answers is not in the document]
Comment (ehaligow) [reply svc-api/handle/chassis.go]: Renamed variables.

File: svc-api/handle/chassis.go [hunk 6]
```diff
@@ -229,3 +230,35 @@ func (chassis *ChassisRPCs) DeleteChassis(ctx iris.Context) {
...
+		SessionToken: ctx.Request().Header.Get("X-Auth-Token"),
+		URL:          ctx.Request().RequestURI,
+		RequestBody:  *requestBody,
+	})
+
+	if rerr != nil {
+		log.Println("RPC error:" + rerr.Error())
```
[the remark this answers is not in the document]
Comment (ehaligow) [reply svc-api/handle/chassis.go]: Done.

File: svc-api/rpc/chassis.go (hunk 1 above)
Comment (ehaligow) [reply svc-api/rpc/chassis.go:120]: Removed "error:" from error below.

File: svc-device-manager/rest/redfish/client.go (hunk 2 above)
Comment (ehaligow) [reply svc-device-manager/rest/redfish/client.go:147]: Done.

File: svc-systems/rpc/chassis.go (hunk 3 above)
Comment (ehaligow) [reply svc-systems/rpc/chassis.go:62]: This method's signature is defined in ChassisServer interface generated by protoc gen go chassis.pb.go, and as an argument it takes context.

**Review by MaciejMis — APPROVED** [review]

**Comment by MaciejMis** [comment]
Accepted on community meeting 23th Jan

**Pull request merged**